patches/minecraft/net/minecraft/client/gui/recipebook/RecipeBookGui.java.patch

```diff
@@ -18,3 +106,63 @@
           this.field_191888_F.field_71466_p.func_78275_b(languagemanager.func_135044_b());
           this.field_191888_F.field_71474_y.func_74303_b();
        }
+@@ -469,4 +496,59 @@
+       }
+ 
+    }
++
++   private void updateButtons() {
++      this.up.visible = this.offset > 0 && this.field_193018_j.size() > 5;
++      this.down.visible = this.field_193018_j.size() > 5 && this.offset < this.field_193018_j.size() - 5;
++   }
++
++   private static class UpDownButton extends net.minecraft.client.gui.widget.button.Button {
```

Comment:
You should move this to another file IMO.
Patches are supposed to be minimal and adding classes is kinda against that

Reply:
Moved UpDownButton to new class